### PR TITLE
get_all_block_producer_seats + get_all_block_producers_ordered

### PR DESCRIPTION
**Pull request by Kouprin**
Fixes https://github.com/nearprotocol/nearcore/issues/1855. Relies on https://github.com/nearprotocol/nearcore/pull/1882.

- discuss with @bowenwang1996 and @SkidanovAlex where seats are necessary
- how to test?

### CODE REVIEW

File: chain/epoch_manager/src/lib.rs [hunk 1]
```diff
@@ -439,7 +439,8 @@ impl EpochManager {
         Ok(epoch_info.validators[seat.tenant as usize].clone())
     }
 
-    pub fn get_all_block_producer_info(
+    /// Returns all unique block producers in current epoch sorted by validator_stake, with indicator on whether they are slashed or not.
```
Comment (bowenwang1996) [inline chain/epoch_manager/src/lib.rs]: It is sorted by account id

Comment (Kouprin) [reply chain/epoch_manager/src/lib.rs]: I checked `proposals_to_epoch_info()`, you're absolutely right.

**Comment by Kouprin** [comment]
> Why do you need `get_all_block_producer_seats`?

I think you mean `get_epoch_block_producer_seats`. You're right, it's overhead. I'll remove it.

**Comment by Kouprin** [comment]
> Why do you need `get_all_block_producer_seats`?

I think `get_epoch_block_producer_seats` may be used instead of `get_epoch_block_producers_ordered` if we understand that we missed seats at some point and need to get them. I removed `get_epoch_block_producer_seats` for now.

**Comment by bowenwang1996** [comment]
> I think `get_epoch_block_producer_seats` may be used instead of `get_epoch_block_producers_ordered` if we understand that we missed seats at some point and need to get them. I removed `get_epoch_block_producer_seats` for now.

I don't quite understand. Is there an example where it is necessary to get all the seats?

**Comment by Kouprin** [comment]
> > I think `get_epoch_block_producer_seats` may be used instead of `get_epoch_block_producers_ordered` if we understand that we missed seats at some point and need to get them. I removed `get_epoch_block_producer_seats` for now.
> 
> I don't quite understand. Is there an example where it is necessary to get all the seats?

Please check `get_part_owner()` and https://github.com/nearprotocol/nearcore/issues/1855 description.

File: chain/epoch_manager/src/lib.rs [hunk 2]
```diff
@@ -439,37 +439,40 @@ impl EpochManager {
...
-        let mut validators: HashSet<AccountId> = HashSet::default();
         for seat in epoch_info.block_producers.iter() {
             let validator_stake = epoch_info.validators[seat.tenant as usize].clone();
-            if !validators.contains(&validator_stake.account_id) {
-                let is_slashed = slashed.contains_key(&validator_stake.account_id);
-                validators.insert(validator_stake.account_id.clone());
-                result.push((validator_stake, is_slashed));
-            }
+            let is_slashed = slashed.contains_key(&validator_stake.account_id);
+            result.push((seat.clone(), validator_stake, is_slashed));
```
Comment (bowenwang1996) [inline chain/epoch_manager/src/lib.rs]: I don't think you need to add `seat` here. We just need `Vec<(ValidatorStake, bool)>` to be returned.

Comment (Kouprin) [reply chain/epoch_manager/src/lib.rs]: ++

**Pull request merged**